apisix/plugins/authz-keycloak.lua

```diff
@@ -0,0 +1,128 @@
+--
+-- Licensed to the Apache Software Foundation (ASF) under one or more
+-- contributor license agreements.  See the NOTICE file distributed with
+-- this work for additional information regarding copyright ownership.
+-- The ASF licenses this file to You under the Apache License, Version 2.0
+-- (the "License"); you may not use this file except in compliance with
+-- the License.  You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+--
+local core     = require("apisix.core")
+local http = require "resty.http"
+local sub_str  = string.sub
+local plugin_name = "authz-keycloak"
+local url = require "net.url"
+local tostring = tostring
+
+
+local schema = {
+    type = "object",
+    properties = {
+        token_endpoint = {type = "string"},
+        permissions = {type = "string"},
+        grant_type = {
+            type = "string",
+            default="urn:ietf:params:oauth:grant-type:uma-ticket"
+        },
+        audience = {type = "string"},
+        timeout = {type = "integer", minimum = 1, default = 3},
```

Comment:
It is recommended to use milliseconds here. So the default value should be  3000 .